clippy_lints/src/trait_bounds.rs

```diff
@@ -70,6 +70,35 @@ declare_clippy_lint! {
     "Check if the same trait bounds are specified twice during a function declaration"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for multiple instances of the same where clause or trait bound
+    ///
+    /// ### Why is this bad?
+    /// Adds to code noise
+    ///
+    /// ### Example
+    /// ```rust
+    /// fn foo<T: Default + Default>(bar: T) {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T: Default>(bar: T) {}
+    /// ```
+    ///
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default + Default {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default {}
+    /// ```
+    #[clippy::version = "1.62.0"]
+    pub REPEATED_WHERE_CLAUSE_OR_TRAIT_BOUND,
+    nursery,
```

Comment:
I fixed the issue in rustc and moved the other lints back to pedantic: https://github.com/rust-lang/rust/pull/96770

So regardless whether this lint is merged with the other lint or kept on its own, it should be in pedantic.

Reply:
Moved it back to pedantic.